### DISCUSSION TITLE
The Pdfalyzer is a PDF internals visualization tool built with PyPDF2 (and a lot of colors)

### DISCUSSION THREAD

**Discussion by michelcrypt4d4mus** [py-pdf/pypdf#1398] [General]
Figured this was worth a post here since @MartinThoma swung by [The Pdfalyzer repo](https://github.com/michelcrypt4d4mus/pdfalyzer) to offer a PR the other day and expressed his appreciation for the project.  I suspect The Pdfalyzer could be quite a useful tool for anyone who has to deal with the internal plumbing of the portable document format in any way.

[Lots of enormous screenshots over at the repo](https://github.com/michelcrypt4d4mus/pdfalyzer#example-output). Also available [on PyPi](https://pypi.org/project/pdfalyzer/).

p.s. thanks for maintaining `PyPDF2`; would not have been able to put this together without it.

**Comment by MartinThoma** [comment]
I've recently added pdfalyzer to https://pypdf2.readthedocs.io/en/latest/meta/faq.html#who-uses-PyPDF2 - Good idea to promote it here as well👍

@pubpub-zz you're digging way deeper into pdf files than I am. Maybe pdfalyzer is useful to you :-)
